### PR TITLE
Removing CHERI-MIPS

### PR DESCRIPTION
**Pull request by bsdjhb**
I have not touched the x86 sketch as the outstanding changes for the x86 chapter would conflict and I would like to merge those first and then revisit x86.

I have mostly not touched the assurance chapter as it is very specific to CHERI-MIPS and L3.

I'm also not sure if the ACK for ETC in the last commit is fully correct or not.

I have added various new jhbnote's that might be worth resolving now if I get some guidance on how to resolve them.

### CODE REVIEW

File: chap-research.tex
Comment (brooksdavis) [inline chap-research.tex:918]: I think we're done so this could be less forward looking?

File: chap-architecture.tex
Comment (brooksdavis) [inline chap-architecture.tex:1805]: I think this is a libcheri thing and should be dropped.

File: chap-architecture.tex
Comment (brooksdavis) [inline chap-architecture.tex:2446]:
```suggestion
In each of the target ISAs (RISC-V and x86-64), we introduce a new
```

File: chap-cheri-riscv.tex
Comment (brooksdavis) [inline chap-cheri-riscv.tex:299]:
```suggestion
This allows capabilities to be embedded within in-memory data structures,
```

File: chap-intro.tex
Comment (brooksdavis) [inline chap-intro.tex:510]:
```suggestion
adaptations of CHERI to ISAs will adopt local instruction-encoding
```
It to me like there's a missing qualifier to "instruction-encoding conventions"

File: chap-sail.tex
Comment (brooksdavis) [inline chap-sail.tex:67]: pre-existing typo
```suggestion
% XXX: for some reason sail generates a link to "to" for ccleartags/cloadtags
```

File: chap-architecture.tex
Comment (bsdjhb) [reply chap-architecture.tex:1805]: It is, though I had just copied this section over with little change from the MIPS chapter.  There are several more libcheri references that need dropping.  For now I just wanted to extricate this content from the MIPS chapter so it wasn't lost.

File: chap-research.tex
Comment (bsdjhb) [reply chap-research.tex:918]: Hmm, I had kept the 'at this time...' clause since it was in the original language, but it could just be dropped.  Not sure if we want some kind of qualification (e.g. "In 2021, we transitioned our research..."

File: chap-intro.tex
Comment (bsdjhb) [reply chap-intro.tex:510]: The qualifier is later in the sentence but it is perhaps a bit awkard to read.  Namely, the "in keeping with their specific flavor and design." at the end.  Maybe I should reword it as:

```
we anticipate that adaptations of CHERI to ISAs will adopt conventions such as instruction-encoding in keeping with their specific flavor and design.
```

File: chap-research.tex
Comment (brooksdavis) [reply chap-research.tex:918]: Sounds good

File: chap-architecture.tex
Comment (brooksdavis) [reply chap-architecture.tex:1805]: Doing this in a later pass seems fine.

**Comment by michael-roe** [comment]
So, I might have an irrational emotional attachment to MIPS, but I don't
really have strong objections to it's removal.

The assurance sections are klnd of important, as they're the high level
overview of the approach we took to formalizing security (later expanded in
in greate detail by various people who were working on machine-checked
proofs).

I could, if this was felt to be a worthwhile activity, make it less
specific to MIPS.

**Comment by bsdjhb** [comment]
I do think having the assurance chapter made less specific to MIPS would be helpful, yes.  I think it's fine if that happens as a separate followup rather than as part of this PR.

**Review by arichardson — COMMENTED** [review]
Haven't reviewed everything in detail but this generally LGTM. One thing that should also be done is changing `update-sail-defs: update-sail-defs-mips update-sail-defs-riscv` to `update-sail-defs: update-sail-defs-riscv` in the Makefile (or completelely removing the MIPS targets.

**Comment by bsdjhb** [comment]
Thanks, I went ahead and removed the build glue entirely.  While I don't have a commit that does it yet, after this lands it probably makes sense to remove the MIPS chapters and the MIPS SAIL files.

File: chap-intro.tex
Comment (arichardson) [inline chap-intro.tex:648]: I think we could keep this sentence, but I also don't mind deleting it.

File: glossary.tex
Comment (arichardson) [inline glossary.tex:777]: I think there is a macro for `Access\_System\_Registers`?

**Review by arichardson — APPROVED** [review]
A few minor suggestions, but also happy to merge this as-is.

**Comment by bsdjhb** [comment]
Now that the earlier x86 changes have landed, I've add a commit to update the x86 chapter as well.

File: app-experimental.tex
Comment (bsdjhb) [reply app-experimental.tex:443]: We don't actually have a description for CLR.C, so that one I've made a `noref`.

File: chap-intro.tex
Comment (bsdjhb) [reply chap-intro.tex:648]: In general I'm removing almost all references to split register files except for a historical note in the rationale chapter.  (I have a followup branch that finishes removing it from CHERI-RISC-V which this PR doesn't yet fully do.)  In that vein I think it makes sense to remove it here.

File: chap-architecture.tex
Comment (bsdjhb) [reply chap-architecture.tex:30]: Hmmm.  Something like that probably is warranted.  I think we still want the strong statement about CHERI-RISC-V in general, but perhaps a qualifier that 32-bit is more experimental while 64-bit is considered mature.

File: glossary.tex
Comment (bsdjhb) [reply glossary.tex]: Yes, I've used it here and in a few other places that spelled it out.

**Comment by arichardson** [comment]
Is anything blocking this from being merged?

**Pull request merged**
